numba/tests/test_array_reductions.py

```diff
@@ -665,6 +668,93 @@ def test_nancumsum_basic(self):
         self.check_cumulative(array_nancumsum)
         self.check_nan_cumulative(array_nancumsum)
 
+    def test_ptp_basic(self):
+        pyfunc = array_ptp_global
+        cfunc = jit(nopython=True)(pyfunc)
+
+        def check(a):
+            expected = pyfunc(a)
+            got = cfunc(a)
+            self.assertPreciseEqual(expected, got)
+
+        def a_variations():
+            yield np.arange(10)
+            yield np.array([-1.1, np.nan, 2.2])
+            yield np.array([-np.inf, 5])
+            yield (4, 2, 5)
+            yield (1,)
+            yield np.full(5, 5)
+            yield [2.2, -2.3, 0.1]
+            a = np.linspace(-10, 10, 16).reshape(4, 2, 2)
```

Comment:
Perhaps also reverse `[::-1]`, and `np.random.shuffle` this to get some randomized values for input?

Reply:
Done